pkg/asset/installconfig/azure/validation.go

```diff
@@ -188,11 +194,15 @@ func validateInstanceTypes(client API, ic *types.InstallConfig) field.ErrorList
 		if ultraSSDCapability == "" {
 			ultraSSDCapability = defaultUltraSSDCapability
 		}
-		if vmNetworkingType == "" {
-			vmNetworkingType = defaultVMNetworkingType
+		capabilities, err = client.GetVMCapabilities(context.TODO(), instanceType, ic.Azure.Region)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(fieldPath.Child("type"), instanceType, err.Error()))
 		}
 		ultraSSDEnabled := strings.EqualFold(ultraSSDCapability, "Enabled")
-		allErrs = append(allErrs, ValidateInstanceType(client, field.NewPath("controlPlane", "platform", "azure"), ic.Azure.Region, instanceType, diskType, controlPlaneReq, ultraSSDEnabled, vmNetworkingType)...)
+		// if capabilities fails to find a valid instance type, we should skip this validation
```

Comment:
The way the code is currently, the validation will also check the compute nodes before returning all the errors found at the end. Maybe we can do the immediate return as a separate PR with the reasoning why?

Reply:
We skip the validation if the instance type is not valid and we add to the list of errors. Because these sections are different in the install config means that we have to check master and compute independently. This was the previously behavior and I followed that pattern.

Reply:
> Oh, I see. If `capabilities == nil` here that means `err != nil` above? this makes sense.

Exactly.